src/backoffice/settings/dashboards/health-check/views/health-check-group.element.ts

```diff
@@ -0,0 +1,204 @@
+import { UUIButtonState } from '@umbraco-ui/uui';
+import { UUITextStyles } from '@umbraco-ui/uui-css/lib';
+import { css, html, nothing } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+
+import { UmbHealthCheckContext } from '../health-check.context';
+import { UmbHealthCheckDashboardContext } from '../health-check-dashboard.context';
+import {
+	HealthCheckAction,
+	HealthCheckGroupWithResult,
+	HealthCheckResource,
+	HealthCheckWithResult,
+	StatusResultType,
+} from '@umbraco-cms/backend-api';
+import { UmbLitElement } from '@umbraco-cms/element';
+import { tryExecuteAndNotify } from '@umbraco-cms/resources';
+import './health-check-action.element';
+
+@customElement('umb-dashboard-health-check-group')
+export class UmbDashboardHealthCheckGroupElement extends UmbLitElement {
+	static styles = [
+		UUITextStyles,
+		css`
+			uui-box {
+				margin-bottom: var(--uui-size-space-5);
+			}
+
+			p {
+				margin: 0;
+			}
+
+			.header {
+				display: flex;
+				justify-content: space-between;
+				align-items: center;
+			}
+
+			.check-results-wrapper {
+				margin-top: var(--uui-size-space-5);
+			}
+
+			.check-results-wrapper .check-result:not(:first-child) {
+				padding-top: var(--uui-size-space-5);
+				border-top: 1px solid var(--uui-color-divider-standalone);
+			}
+
+			.check-results-wrapper uui-button {
+				margin-block-start: 1em;
+			}
+
+			.check-result-description {
+				display: flex;
+			}
+
+			.check-result-description span {
+				width: 36px;
+			}
+
+			uui-icon {
+				vertical-align: sub;
+			}
+		`,
+	];
+
+	@property()
+	groupName!: string;
+
+	@state()
+	private _buttonState: UUIButtonState;
+
+	@state()
+	private _group?: HealthCheckGroupWithResult;
+
+	private _healthCheckContext?: UmbHealthCheckDashboardContext;
+
+	@state()
+	private _checks?: HealthCheckWithResult[] | null;
+
+	@state()
+	private _keyResults?: any;
+
+	private _api?: UmbHealthCheckContext;
+
+	constructor() {
+		super();
+		this.consumeContext<UmbHealthCheckDashboardContext>('umbHealthCheckDashboard', (instance) => {
+			this._healthCheckContext = instance;
+
+			this._api = this._healthCheckContext?.apis.get(this.groupName);
+
+			this._api?.getGroupChecks(this.groupName);
```

Comment:
It looks like we not only _get_ the checks but actually _perform_ the checks when you click on a group:

![image](https://user-images.githubusercontent.com/752371/214583847-2106a869-8c7b-4a78-9f1b-dc0bd50e8d16.png)

Is there a way to rectify that where we still show the checks but do not perform the results before the button "Check" is clicked?

Reply:
P.S. We only show the checks on the second run, when you have clicked the button anyway.

Reply:
Yeah me and @madsrasmussen discussed that when opening a specific group directly via anURL, forexample /health-check/configuration, we wanted to load only the data from configuration.
But there is currently only two APIs available: either we get all checks / data from all the groups, but with no check.results performed, or get a single group with check.results performed.
So the dashboard is currently kind of mimicking the behavior we wanted by deleting the check.results (see health-check.context.ts)
That means that we expecting to have additional api in there future  that can get the group check, without actually performing the check.results.

Another option would be to get all groups without the check.results and only use the part we need for configuration, etc.

Reply:
Correct, this was a workaround to test an alternativ data structure. The current endpoints did too much at the same time. We need to talk to the backend team about the changes. I just totally forgot about it.

Reply:
I think we need to run it by @elit0451 then. It would be good to have an endpoint that just fetches the checks and another that performs the checks and returns the results.